### ISSUE TITLE
[BUG] Not all stations display song info in mplayer/vlc

### ISSUE THREAD

**Issue by Scary-Guy** [coderholic/pyradio#269]
**Describe the bug**
See title

**To Reproduce**
Stations like detroitindustiral.org show the song info.  However most stations I have tried do not do this.  Stations on Laut.fm sanctuaryradio.com and somafm.com do not.

**Expected behavior**
The data is shown on the status line.  I know the stations send out this info too as I can see it in other players (except for the clubmix station on sanctuaryradio which does not send that information out.)

**System (please complete the following information):**
 - OS: LMDE6 (Dabian)
 - **Python** version: 3.11.2-1+b1
 - `pipx` version (if applicable): 1.1.0-1
 - `pipx` installation method: System dependent (I think, not too familiar with pipx and not sure how to check, but I don't think it's applicable here either.)
 - **PyRadio** Version: 0.9.3.11.3
 - Player used: mpv (currently)

**Additional context**
Possible feature upgrade.  Maybe have the status line show as:
```Station: title of station or URL to stream
Title: track info (which may also just be the station ID/url so I'm not sure how well it would work.)```
The reason for this is if there is ever a visualizer or a user is looking at a different page (pretty much just the .csv file list) then the user wont know what station is currently playing if it's not in the stream's info.

**Comment by s-n-g** [comment]
can you provide a list of stations?

Also

> Title: track info (which may also just be the station ID/url so I'm not sure how well it would work.)```
> The reason for this is if there is ever a visualizer or a user is looking at a different page (pretty much just the .csv fille list) then the user wont know what station is currently playing if it's not in the stream's info.

What do you mean exactly?

**Comment by Scary-Guy** [comment]
I provided a list:
"Stations like detroitindustiral.org show the song info. However most stations I have tried do not do this. Stations on Laut.fm sanctuaryradio.com and somafm.com do not."

Also by that I mean that if I have detroitindustrial.org playing and I look at the playlists page I can't see that the current station is DIU in the status line, only the current track that is playing.  This is not such a big deal since most users don't keep the playlists page open all the time (though they could, and I just thought of another potential feature to add where it opens directly to that page instead of just the default playlist.)

**Comment by s-n-g** [comment]
> I provided a list

No you haven't.
I want a list of URLs to test.
I really do not have the time to go hunting for them

> Also by that I mean that if I have detroitindustrial.org playing and I look at the playlists page I can't see that the current station is DIU in the status line, only the current track that is playing. 

Displaying the station name along with the title?
Hm... that could be possible, a config option maybe?

> This is not such a big deal since most users don't keep the playlists page open all the time (though they could, and I just thought of another potential feature to add where it opens directly to that page instead of just the default playlist.)

Do you mean you'd like to have the Playlist mode as the default mode for the application?
Like you always press "o" after opening pyradio?

**Comment by Scary-Guy** [comment]
That's all you needed to say, hang on I'll just paste my favorites
```Detroit Industrial Underground,http://138.197.0.4:8000/stream
Laut.fm/Panorama80,https://stream.laut.fm/panorama80
Laut.fm/Klangwelt,https://stream.laut.fm/klangwelt
Sanctuary Radio Main,http://66.94.113.88:8000/ultra_hq
Sanctuary Radio Retro,http://66.94.113.88:8000/retro_ultra_hq```

The first link works fine and the other four do not.

Also yes, I mean like always having o open.  Then I could use the playlists sort of like directories and sort the entries in them by whatever (url, genre, country etc...) in .csv files dedicated to that.

**Comment by s-n-g** [comment]
I see nothing wrong with them...
All stations show the song title at the bottom bar, and all of them display the station info

Or maybe I haven't understood what you mean?

https://github.com/user-attachments/assets/4b3f36b6-7e3f-4e6f-808c-b0dcd0a33f0a

**Comment by Scary-Guy** [comment]
No, that just means it's working for you and not for me.

Now I just need to figure out why it isn't currently working for me.

**Comment by s-n-g** [comment]
> Also yes, I mean like always having o open.

ok, I suppose that would be possible...
But why would you want to do that?

> Then I could use the playlists sort of like directories and sort the entries in them by whatever (url, genre, country etc...) in .csv files dedicated to that.

You can already do that currently.

The "Playlist mode" is just a transient mode, you just open it to load a different playlist, and get back to the "Main mode" (the list of stations). That's the way the program works...

**Comment by s-n-g** [comment]
> No, that just means it's working for you and not for me.
> 
> Now I just need to figure out why it isn't currently working for me.

Do you have any lua scripts installed for mpv?

**Comment by Scary-Guy** [comment]
Not currently, I'm going to try the other two and see what happens.



> > Also yes, I mean like always having o open.
> 
> ok, I suppose that would be possible... But why would you want to do that?
> 
> > Then I could use the playlists sort of like directories and sort the entries in them by whatever (url, genre, country etc...) in .csv files dedicated to that.
> 
> You can already do that currently.
> 
> The "Playlist mode" is just a transient mode, you just open it to load a different playlist, and get back to the "Main mode" (the list of stations). That's the way the program works...

It prevents me from having to hit o to see them all each time.  Users could still have a favorites file, but it would be displayed along all the others.

I do the same thing in my file system instead of loading up my home (because I don't always want to go into home) so instead of /home/user I just load up / and navigate there as needed (I'm weird like that.) (Well really I use two file browsers as needed to get where I want quickly and now I'm overexplaining again :P )

**Comment by Scary-Guy** [comment]
> > No, that just means it's working for you and not for me.
> > Now I just need to figure out why it isn't currently working for me.
> 
> Do you have any lua scripts installed for mpv?

I just tried with mpv, vlc, and mplayer (which I had to install.) I'm getting no info with any of them except on detroitindustrial.org and I am very confused as to why now.

**Comment by s-n-g** [comment]
I'm in the process of installing lmde on vm
Let me check...

**Comment by Scary-Guy** [comment]
> I'm in the process of installing lmde on vm Let me check...

Wow I super appreciate that.

I just got an error too "mplayer: Player terminated abnormally!"

I was getting a similar error in MPD/NCMPCPP where the start of a song would play and when the track information hit then it would just stop playing.  I had to fix it by removing an option in the mpd.conf as I mentioned over at https://github.com/MusicPlayerDaemon/MPD/discussions/2169 but this is more an mplayer problem than a pyradio problem.  I just felt it warranted a mention.

**Comment by s-n-g** [comment]
So, this is what I have...

- no player can play any of the  Laut.fm stations (due to being on vm?) 
- all players can play both of the Sanctuary stations
- only `mpv` will display titles

https://github.com/user-attachments/assets/0146b914-1591-497c-9a34-cef6264b039a

Although it may not be the best way to test this (being on vm and all), I do have titles with `mpv` and `mplayer` will play both of the Sanctuary stations, even thought it will not display the titles)

**Comment by s-n-g** [comment]
A stupid question, but.. I have to ask: have you installed the multimedia codecs on "Wellcome / First Steps"?

**Comment by Scary-Guy** [comment]
Laut.fm takes longer to connect to but they will eventually load.

I just checked and apparently I didn't have MPV installed so it was using VLC, oops.  Though the track doesn't update on change now so...  new issue!  At least I can reconnect to update it though if I want to know, so that's an improvement.

I never read the first steps, but I did enable the restricted codecs via ```sudo apt install mint-meta-codecs```.
